From 187c75e1dc05069000eb27f3b00e8c38eae57a38 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 28 Apr 2019 22:27:58 +0300
Subject: [PATCH] [SW-0] preboot: use API for skip PCIe init mask

Instead of playing the include game in Makefile, use API function for
receiving the device specific "skip PCIe init" mask.

Note: we still play the include game.

Change-Id: Ibcd94d721b0fa08991fb6340eea082276a34c7c8
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/preboot.c b/common/preboot.c
index 2cc3072..b2d54b0 100644
--- a/common/preboot.c
+++ b/common/preboot.c
@@ -89,7 +89,9 @@
 
 static void preboot_pcie_init(void)
 {
-	if (regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & SKIP_PCIE_INIT) {
+	u32 skip_mask = get_skip_pcie_init_mask();
+
+	if (regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS) & skip_mask) {
 		preboot_puts("PCIe initialization is skipped!\n");
 		return;
 	}
diff --git a/common/preboot.h b/common/preboot.h
index ee3aef2..236e0ed 100644
--- a/common/preboot.h
+++ b/common/preboot.h
@@ -46,6 +46,7 @@
 
 u64 get_cpu_boot_status_reg(void);
 u64 get_upd_pending_sts_reg(void);
+u32 get_skip_pcie_init_mask(void);
 void hl_uart_init(void);
 u32 hl_spi_baudr_get(void);
 
diff --git a/gaudi/gaudi.c b/gaudi/gaudi.c
index 7459bab..61a4598 100644
--- a/gaudi/gaudi.c
+++ b/gaudi/gaudi.c
@@ -42,6 +42,12 @@
 	return mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_3;
 }
 
+u32 get_skip_pcie_init_mask(void)
+{
+	/* Use DumpSel[11] to skip PCIe initialization */
+	return BIT(19);
+}
+
 u64 hl_pll_get_cpu_pll_base_off(void)
 {
 	return mmPSOC_CPU_PLL_BASE;
diff --git a/gaudi/gaudi.h b/gaudi/gaudi.h
index 768149c..8d13e8c 100644
--- a/gaudi/gaudi.h
+++ b/gaudi/gaudi.h
@@ -5,8 +5,6 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-/* Use DumpSel[11] to skip PCIe initialization */
-#define SKIP_PCIE_INIT BIT(19)
 /* Use DumpSel[9] to skip PLL initialization */
 #define SKIP_PLL_INIT BIT(17)
 
diff --git a/goya/goya.c b/goya/goya.c
index 3f00fde..5f14bae 100644
--- a/goya/goya.c
+++ b/goya/goya.c
@@ -168,6 +168,12 @@
 	return mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3;
 }
 
+u32 get_skip_pcie_init_mask(void)
+{
+	/* Use DumpSel[8] to skip PCIe initialization */
+	return BIT(21);
+}
+
 u64 hl_pll_get_cpu_pll_base_off(void)
 {
 	return mmCPU_PLL_BASE;
diff --git a/goya/goya.h b/goya/goya.h
index afdf84c..40c36c0 100644
--- a/goya/goya.h
+++ b/goya/goya.h
@@ -5,8 +5,5 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-/* Use DumpSel[8] to skip PCIe initialization */
-#define SKIP_PCIE_INIT BIT(21)
-
 #define mmCPU_PLL_BASE		0x4A2000
 #define mmIC_PLL_BASE		0x4A3000
